### DISCUSSION TITLE
No additional checks on the source domain?

### DISCUSSION THREAD

**Discussion by marcos-diaz** [plausible/analytics#2125] [Anything Else]
I'm trying Plausible and so far I was able to setup it with no major issue.

But I realised that when I deploy to a temporal staging site (different domain) the metrics are tracked and aggregated only based on the defined `data-domain` attribute? 🤔

That scenario is not a big deal, but made me think... what is preventing some malicious person to send thousands of requests using my same data-domain, and consume my quota? 

Are there really no additional checks other than trusting the client? 

Or is this a problem with no solution due to the nature of how browsers execute code / make cross origin requests?

Or am I misunderstanding something?

**Comment by metmarkosaric** [comment]
Hi @marcos-diaz! Yes, that's correct and it is intentional. Sending traffic stats to any data-domain regardless of what hostname the script is running on is in fact desirable. There were requests to

- test the integration on a dev url
- send traffic from multiple hostnames into one dashboard.
- send traffic from one hostname to multiple separate dashboards

The data-domain attribute provides this flexibility much like the GA tracking code. Plausible is not unique in that sense. With any web analytics tool you can send spam traffic very easily, it's just not something that actually happens. We haven't encountered any issues with this approach so I wouldn't be too worried. If it ever becomes a problem, we can introduce a feature that allows you to verify the hostname.

In terms of excluding your own visits from your staging site: You can exclude yourself [by using a browser extension](https://plausible.io/docs/excluding) or by [setting a localstorage flag](https://plausible.io/docs/excluding) in your browser.

I hope this helps, thanks!
